From a7d0b4290f56d89b48f4d7f245c856652815dbb1 Mon Sep 17 00:00:00 2001
From: Shunqing Chen <csq@rock-chips.com>
Date: Sat, 10 Oct 2020 17:14:34 +0800
Subject: [PATCH] power: rk817-battery: fix dsoc jumps after resume

The units of gap_soc and sleep_soc are different.

Signed-off-by: Shunqing Chen <csq@rock-chips.com>
Change-Id: I42873f1dd4ce0543aa1e89dd91131a5c1fbfa23b
---
 drivers/power/supply/rk817_battery.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/supply/rk817_battery.c b/drivers/power/supply/rk817_battery.c
index e119402a7532..8ad024419d29 100644
--- a/drivers/power/supply/rk817_battery.c
+++ b/drivers/power/supply/rk817_battery.c
@@ -3265,7 +3265,7 @@ static int rk817_bat_sleep_dischrg(struct rk817_battery_device *battery)
 
 		DBG("calib1: rsoc=%d, dsoc=%d, intval=%d\n",
 		    battery->rsoc, battery->dsoc, sleep_soc);
-		if (gap_soc > sleep_soc) {
+		if (gap_soc / 1000 > sleep_soc) {
 			if ((gap_soc - 5000) > (sleep_soc * 2 * 1000))
 				battery->dsoc -= (sleep_soc * 2 * 1000);
 			else
-- 
2.35.3

